src/pudl/extract/eia_bulk_elec.py

```diff
@@ -0,0 +1,104 @@
+"""Module to extract aggregate data from the EIA bulk electricity download.
+
+EIA's bulk electricity data contains 680,000 timeseries. These timeseries contain a
+variety of measures (fuel amount and cost are just two) across multiple levels of
+aggregation, from individual plants to national averages.
+
+The data is formatted as a single 1.1GB text file of line-delimited JSON with one line
+per timeseries. Each JSON structure has two nested levels: the top level contains
+metadata describing the series and the second level (under the "data" heading)
+contains an array of timestamp/value pairs. This structure leads to a natural
+normalization into two tables: one of metadata and one of timeseries. That is the
+format delivered by this module.
+"""
+from io import BytesIO
+from pathlib import Path
+
+import pandas as pd
+
+from pudl.workspace.datastore import Datastore
+
+
+def _filter_for_fuel_receipts_costs_series(df: pd.DataFrame) -> pd.DataFrame:
+    """Pick out the desired data series."""
+    series_to_extract = ["RECEIPTS_BTU", "COST_BTU"]
+    series_pattern = "|".join(series_to_extract)
+    pattern = rf"^ELEC\.(?:{series_pattern})"
+    try:
+        _filter = df.loc[:, "series_id"].str.match(pattern, na=False)
+        return df.loc[_filter, :]
+    except KeyError:
+        return pd.DataFrame()  # empty
```

Comment:
The 1.1GB line-delimited JSON file is read in chunks of 10k lines. This filter is applied to each chunk. Only about 1% of total lines have the keys we're looking for. so most chunks will hit this `KeyError`

Reply:
There are 19k/680k lines that represent objects other than data series, like category definitions or even plot axes. Those objects don't have `series_id` as a field. Because the 1.1GB text file is read chunk-wise into dataframes and filtered with this function, some entire chunks won't have a column called `series_id`

Reply:
Okay that makes sense, but definitely needs some documentation!